pkg/kubelet/kubelet.go

```diff
@@ -2508,15 +2505,16 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName string, lo
 		podStatus = pod.Status
 	}
 
-	if err := kl.validatePodPhase(&podStatus); err != nil {
-		// No log is available if pod is not in a "known" phase (e.g. Unknown).
-		return fmt.Errorf("Pod %q in namespace %q : %v", name, namespace, err)
+	switch podStatus.Phase {
+	case api.PodRunning, api.PodSucceeded, api.PodFailed:
+	default:
+		return fmt.Errorf("pod %q is in an unknown state and logs are not available", pod.Name)
```

Comment:
Ok, I went back and restructured this more, please review again.

1. Phase is not considered - if we can find a container, great, otherwise, we already return an error to the client
2. I've added more error information when a container can't be logged - two most common scenarios for end users are going to be something like "can't pull image" or "image pull backoff", so I explicitly return an error message for those states, and then otherwise print the reason that the container is in waiting state (when there is no previous container).
3. Simplified some of the code to be more readable.